main.go

```diff
@@ -62,17 +67,34 @@ func handler(w http.ResponseWriter, r *http.Request) {
 	clientID := os.Getenv("GITHUB_CLIENT_ID")
 	clientSecret := os.Getenv("GITHUB_CLIENT_SECRET")
 	redirectURL := os.Getenv("GITHUB_REDIRECT_URL")
+	whitelistString := os.Getenv("WHITELIST_REDIRECT_URLS")
 
 	user, err := githubDAO.GetUser(clientID, clientSecret, code, redirectURL)
 	if err != nil {
 		http.Error(w, err.Error(), http.StatusForbidden)
 		opsFailed.Inc()
 		return
 	}
+
+	whitelist := strings.Split(whitelistString, ",")
+	url := "http://localhost:3827"
+	requestedRedirectURL := r.URL.Query().Get("redirect")
+
+	if (len(requestedRedirectURL) > 0) {
+		// check if in whitelist
+		for _, entry := range whitelist {
+			match, _ := regexp.MatchString("^" + entry, requestedRedirectURL)
```

Comment:
I would change to strip spaces.  `strings.TrimSpace(entry)` in case people add spaces in their comma separated list.  Which they will.  Add a failing test in your unit test.  Generally when writing unit tests that simulate human entered data, it's a good time to consider what are the worst things that they could do at this point.

Reply:
yup good catch. I also added a len() check after the trim to ignore blank entries as this was causing it to fail.